site/src/components/home/config/configDialog.jsx

```diff
@@ -0,0 +1,238 @@
+import { useEffect, useMemo, useState } from "react";
+import { Dialog, DialogActions, DialogContent, DialogTitle, Button, Box, TextField, Checkbox, FormControlLabel } from "@mui/material";
+import httpPrefix from "../../../espaddr";
+import PropTypes from "prop-types";
+
+// Base styling for inputs.     
+const textFieldProps = {
+    margin: "dense",
+    fullWidth: true,
+    variant:"standard"
+};
+
+// enum definition: see types.h/SettingSpec.SettingType
+const settingType = {
+    Integer: 0,
+    PositiveBigInteger: 1,
+    Float: 2,
+    Boolean: 3,
+    String: 4,
+    Palette: 5,
+    Color: 6
+};
+
+/**
+ * Creates an Input field based on the settingType provided.
+ * @param {Props} props React Props
+ * @param {Object} props.setting Object defining the setting to load see. types.h/SettingSpec.
+ * @param {Function} props.updateData The callback to provide the change value to the parent dialog.
+ * @param {Function} props.updateError The callback to provide the error state to the parent dialog.
+ * @returns {React.Component} The input field to load into the dialog
+ */
+const ConfigInput = ({setting, updateData, updateError}) => {
+    const [value, setValue] = useState(setting.value);
+    const [error, setError] = useState(false);
+    const [helper, setHelper] = useState(setting.description);
+    const baseProps = {
+        label: setting.friendlyName,
+        id: setting.name,
+        onChange: (e) => setValue(e.target.value),
+        value: value
+    };
+    useEffect(() => {
+        updateError(setting.name, error);
+    }, [setting, updateError, error]);
+    useEffect(() => {
+        if(value !== setting.value) {
+            if(value === '') {
+                updateData(setting.name, undefined);
+            } else {
+                updateData(setting.name, value);
+            }
+        }
+    }, [value, updateData, setting]);
+    const readOnly = !!setting.readOnly;
+    switch(setting.type) {
+    case settingType.Integer:     
+    case settingType.PositiveBigInteger:
+
+        return <TextField
+            {...baseProps}
+            {...textFieldProps}
+            error={error}
+            onError={(e) => updateError(setting.name, e)}
+            helperText={helper}
+            onChange={(e) => {
+                const v = e.target.value;
+                if( v === '' || v< 0 || v> 2**32) {
```

Comment:
Note that "regular" Integers (i.e., not being PositiveBigIntegers) can be negative, and the min/max values the Integer data type can support are correspondingly different (-2,147,483,648 and 2,147,483,647 to be precise).

Reply:
I forgot to come back over this and add in an entry, initially I just started with the switch statement in the order of the type specification. I have added in integer to be a standard text field now.

Reply:
Great, thanks! For the numeric types we may at some point want to implement client-side "bounds checking" based on the minimumValue and/or maximumValue that can be embedded in the setting specs.

Reply:
I agree. At this stage without any known settings to test with I thought it was best just to get something up for people to start using before implementing validation client side too much. 

Ideally long term is like validation on all numerics, and string length either specified in effects or as default values